pages/api/bitbucket/callbacks/install.ts

```diff
@@ -0,0 +1,19 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { publishMessage } from '../../../../utils/pubsub/pubsubClient';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const topicName = 'rtapish-fromserver';
+  const data = {
+    'repository_provider': 'bitbucket',
+    'installation_code': req.query.code};
+  const msgtype = 'install_callback';
+
+  try {
+    await publishMessage(topicName, data, msgtype);
+  } catch (error) {
+    console.error('Error publishing message:', error);
+    res.status(500).json({ error: '<failure>Failed to publish message. Data must be in the form of a Buffer.</failure>' });
```

Comment:
What is the `failure` tag? Never seen this before.

Reply:
Used it without thinking. Will change it as cursory search shows its not a part of w3c standards yet